icu4c/source/i18n/plurrule.cpp

```diff
@@ -1533,20 +1530,42 @@ FixedDecimal::FixedDecimal() {
 
 FixedDecimal::FixedDecimal(const UnicodeString &num, UErrorCode &status) {
     CharString cs;
-    cs.appendInvariantChars(num, status);
+    int32_t parsedExponent = 0;
+
+    int exponentIdx = num.indexOf(LOW_E);
+    if (exponentIdx < 0) {
+        exponentIdx = num.indexOf((UChar) 0x0045); // capital E
+    }
+    if (exponentIdx >= 0) {
+        UnicodeString compactFractionSubstring;
+        compactFractionSubstring.append(num, 0, exponentIdx);
+        cs.appendInvariantChars(compactFractionSubstring, status);
+
+        UnicodeString exponentSubstring;
+        exponentSubstring.append(num, exponentIdx + 1, num.length());
+        std::string exponentCStr;
```

Comment:
This `std::string` is causing the depstest.py errors in Travis CI.  I need to convert the string to a number, and didn't know how best to do that.  There are heavy-handed and circuitous ways of doing it that don't seem reasonable (ex: use the DecimalQuantity constructor that takes in a UnicodeString, like below).

Let me know the preferable way to go about this.

Reply:
@markusicu pointed out `ICU_Utility::parseAsciiInteger`, which works and satisfies the deptest test in Travis CI as well.